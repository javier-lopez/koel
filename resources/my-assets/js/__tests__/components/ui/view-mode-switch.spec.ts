import each from 'jest-each'
import Component from '@/components/ui/view-mode-switch.vue'
import { shallow } from '@/__tests__/adapter'

describe('components/ui/view-mode-switch', () => {
  it('renders properly', () => {
    expect(shallow(Component, {
      propsData: {
        value: 'thumbnails'
      }
    })).toMatchSnapshot()
  })

  each([['thumbnails'], ['list']]).test('emits the "%s" mode value', mode => {
    const wrapper = shallow(Component, {
      propsData: {
        value: 'list'
      }
    })
    expect(wrapper.input(`input[value=${mode}]`).hasEmitted('input', mode)).toBe(true)
  })
})
