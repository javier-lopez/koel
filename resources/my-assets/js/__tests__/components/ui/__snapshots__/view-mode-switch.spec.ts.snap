// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/view-mode-switch renders properly 1`] = `<span class="view-modes"><label title="View as thumbnails" class="thumbnails active"><input type="radio" value="thumbnails" class="hidden"> <i class="fa fa-th-large"></i> <span class="hidden">View as thumbnails</span></label> <label title="View as list" class="list"><input type="radio" value="list" class="hidden"> <i class="fa fa-list"></i> <span class="hidden">View as list</span></label></span>`;
